### PR TITLE
Refactors promises to be more simple to read

### PR DESCRIPTION
**Pull request by ryanthompson591**
The new workflow introduces 2 promises:
1. sessionCreatedPromise_ that resolves when the simid creative initializes a session.
2. resolveInitializationPromise_ that resolves when initialization is resolved.

The workflow should be:
1. Creative creates  a session.
2. Player calls SIMID:player:Init
3. Creative calls SIMID:Player:startCreative

However, these are all asynchronous so they may be called out of order. In order to keep these in order, this sample code uses promises.

Calling Init will 
1. wait for the session to be created (via sessionCreatedPromise_) 
2. then call SIMID:player:init

Calling playAd will:
1. Wait for the creative to respond to the initialization message (via resolveInitializationPromise_)
2. then call SIMID:player:startCreative

### CODE REVIEW

File: examples/player/simid_player.js [hunk 1]
```diff
@@ -65,16 +65,36 @@ class SimidPlayer {
     /**
      * Resolution function for the session created message
```
Comment (damargulis) [inline examples/player/simid_player.js:66]: Add a type here

File: examples/player/simid_player.js [hunk 2]
```diff
@@ -65,16 +65,36 @@ class SimidPlayer {
...
     });
 
+    /**
+     * Resolution function for the ad being initialized.
+     */
+    this.resolveInitializationPromise_ = null;
+
+    /**
+     * Reject function for the ad being initialized.
+     */
```
Comment (damargulis) [inline examples/player/simid_player.js:85]: Types for these also

File: examples/player/simid_player.js (hunk 1 above)
Comment (ryanthompson591) [reply examples/player/simid_player.js:66]: Done here and elsewhere.

File: examples/player/simid_player.js (hunk 2 above)
Comment (ryanthompson591) [reply examples/player/simid_player.js]: Done.

File: examples/player/simid_player.js [hunk 3]
```diff
@@ -132,7 +161,7 @@ class SimidPlayer {
    * @private
    */
   addListeners_() {
-    this.simidProtocol.addListener(ProtocolMessage.CREATE_SESSION, this.sendInitMessage.bind(this));
+    this.simidProtocol.addListener(ProtocolMessage.CREATE_SESSION, this.onSessionCreated_.bind(this));
     this.simidProtocol.addListener(CreativeMessage.REQUEST_FULL_SCREEN, this.onRequestFullScreen.bind(this));
     this.simidProtocol.addListener(CreativeMessage.REQUEST_PLAY, this.onRequestPlay.bind(this));
     this.simidProtocol.addListener(CreativeMessage.REQUEST_PAUSE, this.onRequestPause.bind(this));
```
Comment (Kaleab470) [inline examples/player/simid_player.js:167]: Maybe consider changing these bind statements to ES6 style too

**Pull request merged**